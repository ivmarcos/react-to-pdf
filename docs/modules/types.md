[react-to-pdf](../README.md) / [Modules](../modules.md) / types

# Module: types

## Table of contents

### Type Aliases

- [ConversionOptions](types.md#conversionoptions)
- [DetailedMargin](types.md#detailedmargin)
- [ReactToPDFOptions](types.md#reacttopdfoptions)
- [TargetElementFinder](types.md#targetelementfinder)
- [UsePDFResult](types.md#usepdfresult)

## Type Aliases

### ConversionOptions

Ƭ **ConversionOptions**: `Object`

#### Type declaration

| Name | Type | Description |
| :------ | :------ | :------ |
| `canvas` | `CanvasConversionOptions` | Canvas options |
| `filename?` | `string` | File name of the PDF file if the method select is save, which is the default. Not used for the 'build' and 'open' methods. * |
| `method` | ``"save"`` \| ``"open"`` \| ``"build"`` | Method that will follow to do with the PDF file. The `build` method just returns the PDF instance in the invoked function 'generatePDF' or 'toPDF'. |
| `onCanvas?` | (`canvas`: `HTMLCanvasElement`, `pageNumber`: `number`) => `void` | - |
| `overrides` | { `canvas?`: `Partial`<`Html2CanvasOptions`\> ; `pdf?`: `Partial`<`jsPDFOptions`\>  } | - |
| `overrides.canvas?` | `Partial`<`Html2CanvasOptions`\> | - |
| `overrides.pdf?` | `Partial`<`jsPDFOptions`\> | - |
| `page` | `PageConversionOptions` | Page options |
| `resolution` | [`Resolution`](../enums/constants.Resolution.md) \| `number` | Resolution in a scale where 1 gives a low resolution and possible blurred image, 3 a medium and 10 an extreme quality. The size of the file increases as the resolution is higher. Not recommended to use extreme resolution, e.g '>= 10' for multiple pages otherwise this can make the browser cache hang or crash, due to the size of the image generated for the PDF. |

#### Defined in

types.ts:36

___

### DetailedMargin

Ƭ **DetailedMargin**: `Object`

#### Type declaration

| Name | Type |
| :------ | :------ |
| `bottom` | [`Margin`](../enums/constants.Margin.md) \| `number` |
| `left` | [`Margin`](../enums/constants.Margin.md) \| `number` |
| `right` | [`Margin`](../enums/constants.Margin.md) \| `number` |
| `top` | [`Margin`](../enums/constants.Margin.md) \| `number` |

#### Defined in

types.ts:6

___

### ReactToPDFOptions

Ƭ **ReactToPDFOptions**: `Omit`<`Partial`<[`ConversionOptions`](types.md#conversionoptions)\>, ``"page"`` \| ``"canvas"`` \| ``"overrides"``\> & { `canvas?`: `Partial`<`CanvasConversionOptions`\> ; `overrides?`: `Partial`<[`ConversionOptions`](types.md#conversionoptions)[``"overrides"``]\> ; `page?`: `Partial`<`PageConversionOptions`\>  }

#### Defined in

types.ts:66

___

### TargetElementFinder

Ƭ **TargetElementFinder**: `RefObject`<`HTMLElement`\> \| () => `HTMLElement` \| ``null``

#### Defined in

types.ts:86

___

### UsePDFResult

Ƭ **UsePDFResult**: `Object`

#### Type declaration

| Name | Type | Description |
| :------ | :------ | :------ |
| `targetRef` | `RefObject`<`any`\> | React ref of the target component |
| `toPDF` | (`options?`: [`ReactToPDFOptions`](types.md#reacttopdfoptions)) => `void` | Generates the pdf |

#### Defined in

types.ts:75
