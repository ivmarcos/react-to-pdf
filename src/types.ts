import { MutableRefObject } from "react";
import { jsPDFOptions } from "jspdf";
import { Options as Html2CanvasOptions } from "html2canvas-pro";
import { Margin, Resolution } from "./constants";

export type DetailedMargin = {
  top: Margin | number;
  right: Margin | number;
  bottom: Margin | number;
  left: Margin | number;
};

type PageConversionOptions = {
  /** Margin of the page in MM, defaults to 0. */
  margin: DetailedMargin | Margin | number;
  /** Format of the page (A4, letter), defaults to A4. */
  format: jsPDFOptions["format"];
  /** Orientation of the page (portrait or landscape), defaults to `portrait`. */
  orientation: jsPDFOptions["orientation"];
};

type CanvasConversionOptions = Pick<
  Html2CanvasOptions,
  "useCORS" | "logging"
> & {
  /**
   * Mime type of the canvas captured from the screenshot, defaults to
   * 'image/jpeg' for better size performance.
   * See: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
   */
  mimeType: "image/jpeg" | "image/png";
  /**
   * Quality ratio of the canvas captured from the screenshot
   * See: https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
   */
  qualityRatio: number;
};

export type ConversionOptions = {
  /**
   * File name of the PDF file if the method select is `save`, which is the
   * default. Not used for the `build` and `open` methods. *
   */
  filename?: string;
  /**
   * Method that will follow to do with the PDF file. The `build` method just
   * returns the PDF instance in the invoked function `generatePDF` or `toPDF`.
   * By default is `open`.
   */
  method: "save" | "open" | "build";
  /**
   * Resolution in a scale where 1 gives a low resolution and possible blurred
   * image, 3 a medium and 10 an extreme quality. The size of the file increases
   * as the resolution is higher. Not recommended to use extreme resolution, e.g
   * '>= 10' for multiple pages otherwise this can make the browser cache hang
   * or crash, due to the size of the image generated for the PDF.
   */
  resolution: Resolution | number;
  /** Page options */
  page: PageConversionOptions;
  /** Canvas options */
  canvas: CanvasConversionOptions;
  /** Override values passed for the jsPDF document and html2canvas */
  overrides: {
    /**
     * Override the values passed for the jsPDF instance. See its docs for more details
     * in https://artskydj.github.io/jsPDF/docs/jsPDF.html.
     * */
    pdf?: Partial<jsPDFOptions>;
    /**
     * Override the values passed for the html2canvas function. See its docs
     * for more details in https://html2canvas.hertzen.com/documentation
     * */
    canvas?: Partial<Html2CanvasOptions>;
  };
};

export type Options = Omit<
  Partial<ConversionOptions>,
  "page" | "canvas" | "overrides"
> & {
  page?: Partial<PageConversionOptions>;
  canvas?: Partial<CanvasConversionOptions>;
  overrides?: Partial<ConversionOptions["overrides"]>;
};

export type UsePDFResult = {
  /**
   * React ref of the target element
   */
  targetRef: MutableRefObject<any>;
  /**
   * Generates the pdf
   */
  toPDF: (options?: Options) => void;
};

export type TargetElementFinder =
  | MutableRefObject<any>
  | (() => HTMLElement | null);
