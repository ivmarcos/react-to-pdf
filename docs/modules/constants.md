[react-to-pdf](../README.md) / [Modules](../modules.md) / constants

# Module: constants

## Table of contents

### Enumerations

- [Margin](../enums/constants.Margin.md)
- [Resolution](../enums/constants.Resolution.md)

### Variables

- [DEFAULT\_OPTIONS](constants.md#default_options)
- [MM\_TO\_PX](constants.md#mm_to_px)

## Variables

### DEFAULT\_OPTIONS

• `Const` **DEFAULT\_OPTIONS**: `Readonly`<[`ConversionOptions`](types.md#conversionoptions)\>

#### Defined in

[constants.ts:20](https://github.com/ivmarcos/react-to-pdf/blob/36bd08b/src/constants.ts#L20)

___

### MM\_TO\_PX

• `Const` **MM\_TO\_PX**: ``3.77952755906``

#### Defined in

[constants.ts:3](https://github.com/ivmarcos/react-to-pdf/blob/36bd08b/src/constants.ts#L3)
