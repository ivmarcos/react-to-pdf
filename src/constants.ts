import { ConversionOptions } from "./types";

export const MM_TO_PX = 3.77952755906;

export enum Resolution {
  LOW = 1,
  NORMAL = 2,
  MEDIUM = 3,
  HIGH = 7,
  EXTREME = 12,
}

export enum Margin {
  NONE = 0,
  SMALL = 5,
  MEDIUM = 10,
  LARGE = 25,
}

export const DEFAULT_OPTIONS: Readonly<ConversionOptions> = {
  method: "open",
  resolution: Resolution.MEDIUM,
  page: {
    margin: Margin.NONE,
    format: "A4",
    orientation: "portrait",
  },
  canvas: {
    mimeType: "image/jpeg",
    qualityRatio: 1,
    useCORS: true,
    logging: false,
  },
  overrides: {},
};
