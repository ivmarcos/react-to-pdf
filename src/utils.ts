import { DEFAULT_OPTIONS } from "./constants";
import { ConversionOptions, Options } from "./types";

export const buildConvertOptions = (options?: Options): ConversionOptions => {
  if (!options) {
    return DEFAULT_OPTIONS;
  }
  return {
    ...DEFAULT_OPTIONS,
    ...options,
    canvas: { ...DEFAULT_OPTIONS.canvas, ...options.canvas },
    page: { ...DEFAULT_OPTIONS.page, ...options.page },
  };
};
