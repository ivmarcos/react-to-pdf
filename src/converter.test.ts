/**
 * @jest-environment jsdom
 */

import { Margin, Resolution } from "./constants";
import Converter from "./converter";
import { Options } from "./types";
import { buildConvertOptions } from "./utils";

export const createPageSnapshotObject = (
  converter: InstanceType<typeof Converter>,
  pageNumber: number
) => {
  return {
    pageNumber,
    offsetY: converter.getCanvasOffsetY(pageNumber),
    heightLeft: converter.getCanvasHeightLeft(pageNumber),
    pageCanvasHeight: converter.getCanvasPageHeight(pageNumber),
    pageCanvasWidth: converter.getCanvasPageWidth(),
    canvas: {
      height: converter.canvas.height,
      width: converter.canvas.width,
      originalWidth: converter.canvas.width / converter.getScale(),
      originalHeight: converter.canvas.height / converter.getScale(),
    },
    horizontalFitFactor: converter.getHorizontalFitFactor(),
    scale: converter.getScale(),
    pageWidthMM: converter.getPageWidthMM(),
    pageHeightMM: converter.getPageHeightMM(),
    pageWidth: converter.getPageWidth(),
    pageHeight: converter.getPageHeight(),
    availableHeight: converter.getPageAvailableHeight(),
    availableWidth: converter.getPageAvailableWidth(),
    availableHeightScaled:
      converter.getPageAvailableHeight() * converter.getScale(),
    availableWidthScaled:
      converter.getPageAvailableWidth() * converter.getScale(),
    margin: converter.options.page.margin,
    numberPages: converter.getNumberPages(),
  };
};

const defaultCanvasDimensions = {
  width: 2000,
  height: 10000,
};

const setupConverter = (
  options?: Options,
  canvasDimensions = defaultCanvasDimensions
): InstanceType<typeof Converter> => {
  const canvas = document.createElement("canvas");
  canvas.setAttribute("width", String(canvasDimensions.width));
  canvas.setAttribute("height", String(canvasDimensions.height));
  const convertOptions = buildConvertOptions(options);
  return new Converter(canvas, convertOptions);
};

type CanvasDimensions = {
  width: number;
  height: number;
};

const resolutionsTestSet: number[] = Object.keys(Resolution)
  .filter((key) => isNaN(Number(key)))
  .map((key) => Resolution[key]);
const marginsTestTest: number[] = Object.keys(Margin)
  .filter((key) => isNaN(Number(key)))
  .map((key) => Resolution[key]);

const canvasDimensionsTestSet: CanvasDimensions[] = [
  {
    width: 2000,
    height: 10000,
  },
  {
    width: 20000,
    height: 100,
  },
];

const buildTestSet = () => {
  const testSet: [string, Options, CanvasDimensions][] = [];
  for (const resolution of resolutionsTestSet) {
    for (const margin of marginsTestTest) {
      for (const canvasDimensions of canvasDimensionsTestSet) {
        testSet.push([
          `Resolution: ${resolution}, Margin: ${margin}, Dimensions: ${canvasDimensions.height}x${canvasDimensions.width}`,
          {
            resolution,
            page: {
              margin,
            },
          },
          canvasDimensions,
        ]);
      }
    }
  }
  return testSet;
};

const testSet = buildTestSet();

describe("Converter", () => {
  test.each(testSet)("%s", (_, options, canvasDimensions) => {
    buildTestSet();
    const converter = setupConverter(options, canvasDimensions);
    const numberPages = converter.getNumberPages();
    let pageNumber = 1;
    while (pageNumber <= numberPages) {
      expect(createPageSnapshotObject(converter, pageNumber)).toMatchSnapshot();
      pageNumber += 1;
    }
  });
});
