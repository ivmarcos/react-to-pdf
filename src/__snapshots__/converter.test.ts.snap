// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converter Resolution: 1, Margin: 3, Dimensions: 100x20000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 1099.8428346470896,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 771.0295013133317,
  "canvas": {
    "height": 100,
    "originalHeight": 100,
    "originalWidth": 20000,
    "width": 20000,
  },
  "heightLeft": 100,
  "horizontalFitFactor": 25.939344689059286,
  "margin": 3,
  "numberPages": 1,
  "offsetY": 0,
  "pageCanvasHeight": 100,
  "pageCanvasWidth": 20000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 1,
}
`;

exports[`Converter Resolution: 1, Margin: 3, Dimensions: 10000x2000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 1099.8428346470896,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 771.0295013133317,
  "canvas": {
    "height": 10000,
    "originalHeight": 10000,
    "originalWidth": 2000,
    "width": 2000,
  },
  "heightLeft": 10000,
  "horizontalFitFactor": 2.5939344689059287,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 0,
  "pageCanvasHeight": 2852.9202391702897,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 1,
}
`;

exports[`Converter Resolution: 1, Margin: 3, Dimensions: 10000x2000 2`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 1099.8428346470896,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 771.0295013133317,
  "canvas": {
    "height": 10000,
    "originalHeight": 10000,
    "originalWidth": 2000,
    "width": 2000,
  },
  "heightLeft": 7147.079760829711,
  "horizontalFitFactor": 2.5939344689059287,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 2852.9202391702897,
  "pageCanvasHeight": 2852.9202391702897,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 2,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 1,
}
`;

exports[`Converter Resolution: 1, Margin: 3, Dimensions: 10000x2000 3`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 1099.8428346470896,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 771.0295013133317,
  "canvas": {
    "height": 10000,
    "originalHeight": 10000,
    "originalWidth": 2000,
    "width": 2000,
  },
  "heightLeft": 4294.1595216594205,
  "horizontalFitFactor": 2.5939344689059287,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 5705.8404783405795,
  "pageCanvasHeight": 2852.9202391702897,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 3,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 1,
}
`;

exports[`Converter Resolution: 1, Margin: 3, Dimensions: 10000x2000 4`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 1099.8428346470896,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 771.0295013133317,
  "canvas": {
    "height": 10000,
    "originalHeight": 10000,
    "originalWidth": 2000,
    "width": 2000,
  },
  "heightLeft": 1441.2392824891303,
  "horizontalFitFactor": 2.5939344689059287,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 8558.76071751087,
  "pageCanvasHeight": 1441.2392824891303,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 4,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 1,
}
`;

exports[`Converter Resolution: 2, Margin: 3, Dimensions: 100x20000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 2199.685669294179,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 1542.0590026266634,
  "canvas": {
    "height": 100,
    "originalHeight": 50,
    "originalWidth": 10000,
    "width": 20000,
  },
  "heightLeft": 100,
  "horizontalFitFactor": 12.969672344529643,
  "margin": 3,
  "numberPages": 1,
  "offsetY": 0,
  "pageCanvasHeight": 100,
  "pageCanvasWidth": 20000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 2,
}
`;

exports[`Converter Resolution: 2, Margin: 3, Dimensions: 10000x2000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 2199.685669294179,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 1542.0590026266634,
  "canvas": {
    "height": 10000,
    "originalHeight": 5000,
    "originalWidth": 1000,
    "width": 2000,
  },
  "heightLeft": 10000,
  "horizontalFitFactor": 1.2969672344529644,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 0,
  "pageCanvasHeight": 2852.9202391702897,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 2,
}
`;

exports[`Converter Resolution: 2, Margin: 3, Dimensions: 10000x2000 2`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 2199.685669294179,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 1542.0590026266634,
  "canvas": {
    "height": 10000,
    "originalHeight": 5000,
    "originalWidth": 1000,
    "width": 2000,
  },
  "heightLeft": 7147.079760829711,
  "horizontalFitFactor": 1.2969672344529644,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 2852.9202391702897,
  "pageCanvasHeight": 2852.9202391702897,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 2,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 2,
}
`;

exports[`Converter Resolution: 2, Margin: 3, Dimensions: 10000x2000 3`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 2199.685669294179,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 1542.0590026266634,
  "canvas": {
    "height": 10000,
    "originalHeight": 5000,
    "originalWidth": 1000,
    "width": 2000,
  },
  "heightLeft": 4294.1595216594205,
  "horizontalFitFactor": 1.2969672344529644,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 5705.8404783405795,
  "pageCanvasHeight": 2852.9202391702897,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 3,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 2,
}
`;

exports[`Converter Resolution: 2, Margin: 3, Dimensions: 10000x2000 4`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 2199.685669294179,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 1542.0590026266634,
  "canvas": {
    "height": 10000,
    "originalHeight": 5000,
    "originalWidth": 1000,
    "width": 2000,
  },
  "heightLeft": 1441.2392824891303,
  "horizontalFitFactor": 1.2969672344529644,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 8558.76071751087,
  "pageCanvasHeight": 1441.2392824891303,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 4,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 2,
}
`;

exports[`Converter Resolution: 3, Margin: 3, Dimensions: 100x20000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 3299.528503941269,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 2313.088503939995,
  "canvas": {
    "height": 100,
    "originalHeight": 33.333333333333336,
    "originalWidth": 6666.666666666667,
    "width": 20000,
  },
  "heightLeft": 100,
  "horizontalFitFactor": 8.64644822968643,
  "margin": 3,
  "numberPages": 1,
  "offsetY": 0,
  "pageCanvasHeight": 100,
  "pageCanvasWidth": 20000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 3,
}
`;

exports[`Converter Resolution: 3, Margin: 3, Dimensions: 10000x2000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 3299.528503941269,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 2313.088503939995,
  "canvas": {
    "height": 10000,
    "originalHeight": 3333.3333333333335,
    "originalWidth": 666.6666666666666,
    "width": 2000,
  },
  "heightLeft": 10000,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 0,
  "pageCanvasHeight": 3299.528503941269,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 3,
}
`;

exports[`Converter Resolution: 3, Margin: 3, Dimensions: 10000x2000 2`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 3299.528503941269,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 2313.088503939995,
  "canvas": {
    "height": 10000,
    "originalHeight": 3333.3333333333335,
    "originalWidth": 666.6666666666666,
    "width": 2000,
  },
  "heightLeft": 6700.471496058732,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 3299.528503941269,
  "pageCanvasHeight": 3299.528503941269,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 2,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 3,
}
`;

exports[`Converter Resolution: 3, Margin: 3, Dimensions: 10000x2000 3`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 3299.528503941269,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 2313.088503939995,
  "canvas": {
    "height": 10000,
    "originalHeight": 3333.3333333333335,
    "originalWidth": 666.6666666666666,
    "width": 2000,
  },
  "heightLeft": 3400.9429921174624,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 6599.057007882538,
  "pageCanvasHeight": 3299.528503941269,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 3,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 3,
}
`;

exports[`Converter Resolution: 3, Margin: 3, Dimensions: 10000x2000 4`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 3299.528503941269,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 2313.088503939995,
  "canvas": {
    "height": 10000,
    "originalHeight": 3333.3333333333335,
    "originalWidth": 666.6666666666666,
    "width": 2000,
  },
  "heightLeft": 101.41448817619312,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 4,
  "offsetY": 9898.585511823807,
  "pageCanvasHeight": 101.41448817619312,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 4,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 3,
}
`;

exports[`Converter Resolution: 7, Margin: 3, Dimensions: 100x20000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 7698.899842529627,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 5397.206509193322,
  "canvas": {
    "height": 100,
    "originalHeight": 14.285714285714286,
    "originalWidth": 2857.1428571428573,
    "width": 20000,
  },
  "heightLeft": 100,
  "horizontalFitFactor": 3.7056206698656124,
  "margin": 3,
  "numberPages": 1,
  "offsetY": 0,
  "pageCanvasHeight": 100,
  "pageCanvasWidth": 20000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 7,
}
`;

exports[`Converter Resolution: 7, Margin: 3, Dimensions: 10000x2000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 7698.899842529627,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 5397.206509193322,
  "canvas": {
    "height": 10000,
    "originalHeight": 1428.5714285714287,
    "originalWidth": 285.7142857142857,
    "width": 2000,
  },
  "heightLeft": 10000,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 2,
  "offsetY": 0,
  "pageCanvasHeight": 7698.899842529627,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 7,
}
`;

exports[`Converter Resolution: 7, Margin: 3, Dimensions: 10000x2000 2`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 7698.899842529627,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 5397.206509193322,
  "canvas": {
    "height": 10000,
    "originalHeight": 1428.5714285714287,
    "originalWidth": 285.7142857142857,
    "width": 2000,
  },
  "heightLeft": 2301.1001574703732,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 2,
  "offsetY": 7698.899842529627,
  "pageCanvasHeight": 2301.1001574703732,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 2,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 7,
}
`;

exports[`Converter Resolution: 12, Margin: 3, Dimensions: 100x20000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 13198.114015765075,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 9252.35401575998,
  "canvas": {
    "height": 100,
    "originalHeight": 8.333333333333334,
    "originalWidth": 1666.6666666666667,
    "width": 20000,
  },
  "heightLeft": 100,
  "horizontalFitFactor": 2.1616120574216073,
  "margin": 3,
  "numberPages": 1,
  "offsetY": 0,
  "pageCanvasHeight": 100,
  "pageCanvasWidth": 20000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 12,
}
`;

exports[`Converter Resolution: 12, Margin: 3, Dimensions: 10000x2000 1`] = `
{
  "availableHeight": 1099.8428346470896,
  "availableHeightScaled": 13198.114015765075,
  "availableWidth": 771.0295013133317,
  "availableWidthScaled": 9252.35401575998,
  "canvas": {
    "height": 10000,
    "originalHeight": 833.3333333333334,
    "originalWidth": 166.66666666666666,
    "width": 2000,
  },
  "heightLeft": 10000,
  "horizontalFitFactor": 1,
  "margin": 3,
  "numberPages": 1,
  "offsetY": 0,
  "pageCanvasHeight": 10000,
  "pageCanvasWidth": 2000,
  "pageHeight": 1122.5200000014497,
  "pageHeightMM": 297.0000833333333,
  "pageNumber": 1,
  "pageWidth": 793.7066666676917,
  "pageWidthMM": 210.0015555555555,
  "scale": 12,
}
`;
